### DISCUSSION TITLE
Layer data as object property not working

### DISCUSSION THREAD

**Discussion by Doddle-BE** [visgl/deck.gl#7070] [Q&A]
I'm rendering a new Text Layer.
The data for this layer is stored (as array) in my state (this.state.data)

When I try to assign the data-property of my layer with my data property of the state, it isn't working.
I first have to make a "copy" of my array to get this working.

Isn't working:
```
new TextLayer({
      id: "text-numbers",
      data: this.state.data,
      getText: d => d.text
})
```

This works:
```
new TextLayer({
      id: "text-numbers",
      data: [...this.state.data],
      getText: d => d.text
})
```

Can someone explain why Deck.gl cannot work with the array directly coming from the state?

**Comment by mcost45** [comment]
It's difficult to deep diff objects with good performance for general usage. So instead a layer determines if it needs to update only on shallow changes, I.e when the data objects aren't equal.

https://deck.gl/docs/developer-guide/using-layers#why-doesnt-my-layer-update-with-new-props
